File: Week02/LeetCode-0220.cpp
class Solution {
public:
    bool containsNearbyAlmostDuplicate(vector<int>& nums, int k, int t) {
        for (int i=0; i<(int)nums.size()-1; i++){
            for (int j=i+1; j<std::min(i+k+1, (int)nums.size()); j++)
                if (std::abs((int64_t)nums[i]-(int64_t)nums[j]) <= t)
                    return true;
        } // end-for        
        return false;        
    } // end-containsNearbyAlmostDuplicate
};

File: Week02/LeetCode-0220.py
class Solution:
    def containsNearbyAlmostDuplicate(self, nums: List[int], k: int, t: int) -> bool:
        """
        O(N^2) algorithm: Gets Time Limit Exceeded
        """
        for i in range(len(nums)-1):
            for j in range(i+1, min(i+k+1, len(nums))):
                if abs(nums[i]-nums[j]) <= t:
                    return True        
        return False        